src/viem/prepareTransactions.ts

```diff
@@ -85,33 +85,33 @@ export async function tryEstimateTransaction({
   baseTransaction,
   maxFeePerGas,
   maxPriorityFeePerGas,
-  feeCurrencySymbol,
-  feeCurrencyAddress,
+  feeCurrency,
 }: {
   client: Client
   baseTransaction: TransactionRequest
   maxFeePerGas: bigint
   maxPriorityFeePerGas?: bigint
-  feeCurrencySymbol: string
-  feeCurrencyAddress?: Address
+  feeCurrency: TokenBalance
 }) {
   const tx = {
     ...baseTransaction,
     maxFeePerGas,
     maxPriorityFeePerGas,
     // Don't include the feeCurrency field if not present.
     // See https://github.com/wagmi-dev/viem/blob/e0149711da5894ac5f0719414b4ecc06ccaecb7b/src/chains/celo/serializers.ts#L164-L168
-    ...(feeCurrencyAddress && { feeCurrency: feeCurrencyAddress }),
+    ...(!feeCurrency.isNative &&
+      feeCurrency.address && { feeCurrency: feeCurrency.address as Address }),
   }
 
-  // TODO maybe cache this? and add static padding when using non-native fee currency
+  // TODO maybe cache this?
   try {
-    tx.gas = await estimateGas(client, {
+    const estimatedGas = await estimateGas(client, {
       ...(tx as TransactionRequestBase),
       account: tx.from,
     })
+    tx.gas = estimatedGas + BigInt(feeCurrency.isNative ? 0 : STATIC_GAS_PADDING)
```

Comment:
Sorry the comment may have been a bit misleading.
The padding would be necessary only if caching this value and the initial value was without a feeCurrency set.

We don't need the padding here if we call `estimateGas` with the `feeCurrency` set.